src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -326,70 +333,106 @@ export function openFacilityPageV2(page, uiSchema, schema) {
+        requestFacilityIds = criteria[1]
+          ?.filter(facility =>
+            facility?.requestSettings?.some(
+              setting => setting.id === typeOfCareId,
+            ),
+          )
+          ?.map(facility => facility.id);
+
+        const uniqueIds = Array.from(
+          new Set([...directFacilityIds, ...requestFacilityIds]),
+        );
+
+        // The above API calls only return the ids. Make an additional
+        // call to getLocations so we can get additional details such
+        // as name, address, coordinates, etc.
+        if (uniqueIds.length) {
+          typeOfCareFacilities = await getLocations({
+            facilityIds: uniqueIds,
           });
+
+          // Update the retrieved locations with requestSupported and
+          // directSchedulingSupported, as well as replace IDs for dev/staging
+          typeOfCareFacilities = typeOfCareFacilities?.map(location =>
+            setSupportedSchedulingMethods({
+              location,
+              requestFacilityIds,
+              directFacilityIds,
+            }),
+          );
         }
-      } catch (err) {
-        dispatch({ type: FORM_FETCH_PARENT_FACILITIES_FAILED });
       }
 
-      locations = newAppointment.facilities[typeOfCareId] || null;
+      const canShowFacilities = !!parentId || parentFacilities?.length === 1;
 
-      if (parentFacilities?.length && !locations) {
-        dispatch({
-          type: FORM_FETCH_CHILD_FACILITIES,
-        });
+      if (canShowFacilities && !parentId) {
+        parentId = parentFacilities[0].id;
+      }
 
-        // Fetch locations for each parent facility
-        const responses = await Promise.all(
-          parentFacilities.map(parent => {
-            const parentId = parent.id;
-            const siteId = parseFakeFHIRId(
-              getIdOfRootOrganization(parentFacilities, parentId),
-            );
-            return getSupportedHealthcareServicesAndLocations({
-              siteId,
-              parentId,
-              typeOfCareId,
-              useVSP,
-            });
-          }),
+      if (parentId) {
+        siteId = parseFakeFHIRId(
+          getIdOfRootOrganization(parentFacilities, parentId),
         );
-
-        locations = [].concat(...responses.map(r => r?.locations || []));
       }
 
       // If we have an already selected location or only have a single location
       // fetch eligbility data immediately
-      const eligibilityDataNeeded = !!locationId || locations?.length === 1;
+      const eligibilityDataNeeded =
+        !!facilityId || typeOfCareFacilities?.length === 1;
 
-      if (eligibilityDataNeeded && !locationId) {
-        locationId = locations[0].id;
+      if (eligibilityDataNeeded && !facilityId) {
+        facilityId = typeOfCareFacilities[0].id;
       }
 
-      if (!locations?.length) {
+      if (!typeOfCareFacilities?.length) {
         parentFacilities.forEach(p => {
```

Comment:
Instead of this loop, could we leave the facilityId off? Multiple events like this might be confusing.

Reply:
Removed the forEach loop and will only fire this for `parentFacilities[0]`.